Caltech-IPAC/firefly
# DM-8010: Add LSSTMetaSearch and LSSTCatalogSearch processors

To test it,

Check it out and build. 
Start firefly view, then enter the following info based on the method:
  Cone search:  
     UserTargetPt: 9.49,-1.24
     Radius: 0.01 degree
 Box Search:
     UserTargetPt: 9.49,-1.24
     Radius: 0.01 degree
 Elliptical Search:
    9.49, -1.24
   Radius: 0.01 degree
Polygon Search:
    9.5  -1.2, 9.6 -1.24, 9.4 -1.24

Known issue:
  When the degree unit is selected, it still passed as arcsec 
  When all columns are selected by default, it passed null

Suggest:
  UI passes the unit, the server side handles the conversion based on the search methods.

Thanks for reviewing it.

[11/1/16 at 5:37PM]
After discussing with Cindy and after her changes, the following changes are made:
   1.  If the UI sends a null for selcol, all columns are selected.
   2.  For radius UI, he "degree" unit is used  The server side converts the unit based on the 
       search methods.
   3.  Datatypes inconsistence (for the same column):
        meta search (LSSTMetaSearch): select SHOW COLUMNS 
               return double, boolean
       catalog search (LSSTCatalogSearch): select * (select id, parent, ...)
               return float for double 
               return null for boolean

        When the DataType is created, if the type is no null, the type in catalog search is always
       used.  If it is null, the type in meta search is used.
   4. Add throw Exception for  LSSTCatalogSearch.  LSSTMetaSearch had the exception handled.
